torch/fx/experimental/proxy_tensor.py

```diff
@@ -757,6 +759,108 @@ def disable_autocast_cache():
+        for name, mod in self.scope_root.named_modules(remove_duplicate=False):
+            self.module_id_cache[id(mod)].append(name)
+
+        self_ = self
+
+        class AttrProxy:
+            def __init__(self, base, path):
+                self.__class__ = type(
+                    base.__class__.__name__,
+                    (self.__class__, base.__class__),
+                    {},
+                )
+                self.__dict__ = base.__dict__
+                self_.proxy_paths[self] = path
+                self_.proxy_modules[self] = base
+
+            def __getattr__(self, name):
+                assert isinstance(self, torch.nn.Module)
+                attr_val = super().__getattr__(name)
+                if isinstance(attr_val, AttrProxy):
+                    attr_val = self_.proxy_modules[attr_val]
+                elif not isinstance(attr_val, torch.nn.Module):
+                    return attr_val
+                return AttrProxy(attr_val, self_.proxy_paths[self] + "." + name)
+
+            @property
+            def _modules(self):
+                assert "_modules" in self.__dict__
+                submodules = self.__dict__["_modules"]
+                assert isinstance(submodules, dict)
+                return {
+                    key: AttrProxy(value, self_.proxy_paths[self] + "." + str(key))
+                    for key, value in submodules.items()
+                }
+
+        self.proxy_type = AttrProxy
+        self.skip_list = set()
+
+    def path_of_module(self, mod: torch.nn.Module) -> str:
+        """
+        Use tracked access path during tracing instead of the default BFS behavior.
+        Still use all the possible module paths to verify the result.
+        """
+        if mod is self.scope_root:
+            return ""
+
+        if isinstance(mod, self.proxy_type):
+            module_id = id(self.proxy_modules[mod])
+            results = self.module_id_cache[module_id]
+            return self.proxy_paths[mod]
+
+        return Tracer.path_of_module(self, mod)
+
+    def getattr(self, attr, attr_val, parameter_proxy_cache):
+        if not isinstance(attr_val, torch.nn.Module):
+            return super().getattr(attr, attr_val, parameter_proxy_cache)
+        return self.proxy_type(attr_val, attr)
+
+    def call_module(self, m, forward, args, kwargs):
+        """PythonKeyTracer overrides call_module to avoid the scope handling,
+        but we actually want it.
+        """
+        from torch._dynamo import OptimizedModule
+        # When we call torch.compile inside HOO, we will end up
+        # invoking a module that is not registered on the root.
+        # So we just temporarily register it there so that tracer
+        # goes through. I think this is ok because make_fx traces
+        # the callable that is not the actual user supplied callable,
+        # so it doesn't matter that we are adding more modules.
+        if isinstance(m, OptimizedModule):
```

Comment:
this is bit sus

Reply:
Can you add a unit test of running `make_fx(f)` where `f._orig_mod` is set, and `f` calls `torch.cond` (you can use an expecttest to assert on what the captured graph looks like). That would make it clearer what the intended behavior is

Reply:
Added a test on test_export.py